File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for (int i=0; i<nums.size(); i++)
        {
            int comp = target - nums[i];
            if (hash.find(comp) != hash.end() )
                return {i, hash[comp]};
            hash[nums[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> remain = new HashMap<>();

        for (int i=0; i<nums.length; i++) {
            int subtract = target - nums[i];
            if (remain.containsKey(subtract)) {
                return new int[]{i, remain.get(subtract)};
            }
            remain.put(nums[i],i);
        }

        return new int[]{};
    }
}